cedar-lean/Cedar/Validation/Typechecker.lean

```diff
@@ -185,13 +185,19 @@ def hasAttrInRecord (rty : RecordType) (x : Expr) (a : Attr) (c : Capabilities)
     else ok (.bool .anyBool) (Capabilities.singleton x a)
   | .none     => ok (.bool .ff)
 
+def actionType? (ety : EntityType) (acts: ActionSchema) : Bool :=
+  acts.keys.any (EntityUID.ty · == ety)
```

Comment:
this isn't the definition of `actionType?` I was intuitively expecting, which would be something like "see if the last component of the entity type is the string `Action`".  Is there a reason to prefer one definition or the other?

Reply:
We made the same decision for [`actionUID?`](https://github.com/cedar-policy/cedar-spec/blob/6d98a57f74bdc9e694997fabdfb402902b84775e/cedar-lean/Cedar/Validation/Typechecker.lean#L135). The rationale was that an action should really be defined as something that is in the action schema, rather than an entity with a type name "Action". This is a different approach from the Rust, but it makes the formalization cleaner. Will defer further argument to @emina.

Reply:
I'm good with that, that is a reasonable explanation.  And difftesting ensures the validators behave equivalently anyway